faunadb-java-dsl/src/main/java/com/faunadb/client/types/Codec.java

```diff
@@ -167,4 +229,172 @@ public R apply(T input) {
     };
   }
 
+  static <I extends O, O> Function<I, O> cast() {
```

Comment:
`cast` is a little vague from the call site. I'd specifically state that this is a `upcast`.